### PR TITLE
adding feature for incremental target data

### PR DESCRIPTION
**Pull request by orvindemsy**
1. what to name variabel for two data, for now I just leave it at Xt1 and Xt2
2. what to name the methods, I'm thinking of using `fit_tr` for Xs, Xt1, and `fit_te`  for Xt2, but doubt the term `tr` (train) and`te` (test) suitable for TCA?
3. I add data folder as well so code can be run directly, because for me the first time I saw this repo I I have to look around different repo to get those, amazon, webcam, dslr data, the baidu link you're referring to in readme doesn't work either for me.

### CODE REVIEW

**Comment by jindongwang** [comment]
Hi @orvindemsy , thanks for this PR. As you mentioned, the dataset download link is hard to use for many people. So I update them to two net disks: Jianguoyun or MEGA, both can be reached easily. Check this address: https://github.com/jindongwang/transferlearning/tree/master/data#office-caltech10
image

For the method names, I suggest you name them `fit_old` and `fit_new` for easy understanding.
I'll check the coder after work. Thanks for your contribution!

**Comment by orvindemsy** [comment]
I didn't know that that link existed, thanks too for kindly providing link on every answer! That really helps. Okay then, will wait for your comment on my code

**Review by jindongwang — CHANGES_REQUESTED** [review]
1. We should avoid modifying the existing TCA fit and fit_predict functions since they are commonly used by others. Instead, add a new function to adapt to new target data is enough.
2. The .mat data should not be added in the repo since they are binary files that will not incrementally tracked by git, so the whole repo will become extremely large.

File: code/traditional/TCA/TCA.py [hunk 1]
```diff
@@ -61,13 +65,20 @@ def fit(self, Xs, Xt):
...
+        # Projecting to latent space
         A = V[:, ind[:self.dim]]
         Z = np.dot(A.T, K)
         Z /= np.linalg.norm(Z, axis=0)
-        Xs_new, Xt_new = Z[:, :ns].T, Z[:, ns:].T
-        return Xs_new, Xt_new
+        
+        Xs_new, Xt1_new = Z[:, :ns].T, Z[:, ns:].T
+        
+        self.W_proj = A
```
Comment (jindongwang) [inline code/traditional/TCA/TCA.py:76]: Maybe it is enough to only retain the W_proj?

Comment (orvindemsy) [reply code/traditional/TCA/TCA.py:76]: sorry don't get it, let me confirm the W_proj is A right? so if you say not to change the .fit method, I should create another method again that build W_proj and/or projects it into Xt2, like 
1. `fit_new(Xs, Xt1, Xt2) -> Xt2_new, W_proj`
or separate them?
1. `fit_wproj (Xs, Xt1) -> W_proj`
2. `fit_new(Xt2) -> Xt2_new`

Comment (jindongwang) [reply code/traditional/TCA/TCA.py:76]: Yes. My key point is not to modify the existing TCA functions (interfaces) but try to add a new function for new target data. That's all.

Comment (orvindemsy) [reply code/traditional/TCA/TCA.py:76]: all right, working on it.

File: code/traditional/TCA/TCA.py [hunk 2]
```diff
@@ -65,20 +61,14 @@ def fit(self, Xs, Xt1):
...
-        self.W_proj = A
-        self.X_new_src = X
-        
-        return Xs_new, Xt1_new
+        Xs_new, Xt_new = Z[:, :ns].T, Z[:, ns:].T
+        return Xs_new, Xt_new
 
-    def fit_predict(self, Xs, Ys, Xt1, Yt1):
+    def fit_predict(self, Xs, Ys, Xt, Yt):
         '''
```
Comment (orvindemsy) [inline code/traditional/TCA/TCA.py]: as per requested I convert all `fit` and `fit_predict` methods and all tis variable name to original, let me know if I miss something

File: code/traditional/TCA/TCA.py [hunk 3]
```diff
@@ -87,50 +77,62 @@ def fit_predict(self, Xs, Ys, Xt1, Yt1):
...
         
         # New target features
-        Xt2_new = K @ self.W_proj
+        Xt2_new = K @ A
         
         return Xt2_new
     
-    def fit_predict_target(self, Xt2, Yt2):
+    def fit_new_predict(self, Xt, Xs, Ys, Xt2, Yt2):
         '''
```
Comment (orvindemsy) [inline code/traditional/TCA/TCA.py:121]: I think it got quite messy here, so these are parameters requires by `fit_new_predict` :
- `Xs` and `Xt`: to be passed into `fit` to create `Xs_new` (ignore `Xt_new`) then these two are also passed into `fit_new` to create `Xt2_new`
- `Ys`: to be paired with `Xs_new` to fit clf
- `Xt2 and Yt2`:  to create `Xt2_new` and also to evaluate on clf

Also I got trouble finding a good name for `Xt2` is it okay enough, any suggestion?
To be consistent with previous code, all variables followed by `new` are transformed feature by TCA.
Sorry to update this on weekend, it's okay you can reply me on weekday or anytime that suits you.

File: code/traditional/TCA/TCA.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def fit(self, Xs, Xt):
         X = np.hstack((Xs.T, Xt.T))
         X /= np.linalg.norm(X, axis=0)
         m, n = X.shape
-        ns, nt = len(Xs), len(Xt)
+        ns, nt = len(Xs), len(Xt1)
```
Comment (orvindemsy) [inline code/traditional/TCA/TCA.py:54]: self-reminder: here I `Xt1` should be `Xt`, correct in the next commit.

**Comment by jindongwang** [comment]
@orvindemsy Saw your changes. Looks wonderful. Is it OK to merge your commits now, or you want to modify or test more?

**Comment by orvindemsy** [comment]
@jindongwang  Thanks!! last commit, Just minor changes:
- `Xt` to `Xt1` in `.fit`
- Rename `.fit_new_predict` to `fit_predict_new` to keep name consistency to old `fit_predict`
- Edit docs on `fit_new` and `fit_predict_new`, to avoid confusion (I hope so)

If you found it okay, then you can merge it.

**Pull request merged**